registry_module.go

```diff
@@ -124,18 +124,20 @@ type CommitList struct {
 
 // RegistryModule represents a registry module
 type RegistryModule struct {
-	ID              string                          `jsonapi:"primary,registry-modules"`
-	Name            string                          `jsonapi:"attr,name"`
-	Provider        string                          `jsonapi:"attr,provider"`
-	RegistryName    RegistryName                    `jsonapi:"attr,registry-name"`
-	Namespace       string                          `jsonapi:"attr,namespace"`
-	NoCode          bool                            `jsonapi:"attr,no-code"`
-	Permissions     *RegistryModulePermissions      `jsonapi:"attr,permissions"`
-	Status          RegistryModuleStatus            `jsonapi:"attr,status"`
-	VCSRepo         *VCSRepo                        `jsonapi:"attr,vcs-repo"`
-	VersionStatuses []RegistryModuleVersionStatuses `jsonapi:"attr,version-statuses"`
-	CreatedAt       string                          `jsonapi:"attr,created-at"`
-	UpdatedAt       string                          `jsonapi:"attr,updated-at"`
+	ID                  string                          `jsonapi:"primary,registry-modules"`
+	Name                string                          `jsonapi:"attr,name"`
+	Provider            string                          `jsonapi:"attr,provider"`
+	RegistryName        RegistryName                    `jsonapi:"attr,registry-name"`
+	Namespace           string                          `jsonapi:"attr,namespace"`
+	NoCode              bool                            `jsonapi:"attr,no-code"`
+	Permissions         *RegistryModulePermissions      `jsonapi:"attr,permissions"`
+	PublishingMechanism string                          `jsonapi:"attr,publishing-mechanism"`
```

Comment:
Within the client, the traditional pattern for "magic string" API fields (where logically speaking it's an enum, it's just that none of our languages handle enums very usefully) is to define a new string type and then define constants for all of the legal values of the field. See the RegistryModuleStatus type right here in this file for an example. 

I'd like to see that for this field, unless there's a good reason to do something different.